File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/IncludedSignatureOutputTests.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.math.BigDecimal;
import org.junit.jupiter.api.Test;

class IncludedSignatureOutputTests extends CqlTestBase {

    @Test
    void evaluate() {
        var results = engine.evaluate(toElmIdentifier("IncludedSignatureOutputTests"));

        Object value = results.forExpression("TestOverload").value();
        assertThat(value, is(5));

        value = results.forExpression("TestOverloadOneInt").value();
        assertThat(value, is(1));

        value = results.forExpression("TestOverloadOneDecimal").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadTwoInts").value();
        assertThat(value, is(1));

        value = results.forExpression("TestOverloadTwoDecimals").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadOneIntOneDecimal").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadOneIntTwoDecimal").value();
        assertThat(value, is(1));
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/execution/IncludedSignatureOutputTests.kt
package org.opencds.cqf.cql.engine.execution

import kotlin.test.Test
import kotlin.test.assertEquals
import org.cqframework.cql.shared.BigDecimal
import org.opencds.cqf.cql.engine.runtime.toCqlDecimal
import org.opencds.cqf.cql.engine.runtime.toCqlInteger

internal class IncludedSignatureOutputTests : CqlTestBase() {
    @Test
    fun evaluate() {
        val results = engine.evaluate { library("IncludedSignatureOutputTests") }.onlyResultOrThrow

        var value = results["TestOverload"]!!.value
        assertEquals(5.toCqlInteger(), value)

        value = results["TestOverloadOneInt"]!!.value
        assertEquals(1.toCqlInteger(), value)

        value = results["TestOverloadOneDecimal"]!!.value
        assertEquals(BigDecimal("2.0").toCqlDecimal(), value)

        value = results["TestOverloadTwoInts"]!!.value
        assertEquals(1.toCqlInteger(), value)

        value = results["TestOverloadTwoDecimals"]!!.value
        assertEquals(BigDecimal("2.0").toCqlDecimal(), value)

        value = results["TestOverloadOneIntOneDecimal"]!!.value
        assertEquals(BigDecimal("2.0").toCqlDecimal(), value)

        value = results["TestOverloadOneIntTwoDecimal"]!!.value
        assertEquals(1.toCqlInteger(), value)
    }
}
